src/lib/autoinstall/autosetup_helpers.rb

```diff
@@ -89,8 +89,15 @@ def modified_profile?
     def suse_register
       return true unless registration_module_available? # do nothing
 
+      register_section = Yast::Profile.current.fetch_as_hash(REGISTER_SECTION, nil)
+      disabled_registration = (register_section || {})["do_registration"] == false
```

Comment:
Because that was the original behavior just adapted it using fetch_as_hash to not use current directly as recommended, there is a check below that check if the section was present or not. So, if I maintain it with the default I will need to modify the if condition later.

Reply:
I don't see the difference for the below check. If it is present and explicitly disable (do_registration == false) it is going to be removed. If it is not present, register_section["do_registration"] != false (because it is nil).

Anyway, it was just a question. A nitpicking question, maybe. It is a merge PR, so you can go ahead and forget this comment :)

Reply:
Thnx to you for checking the logic ;)